### DISCUSSION TITLE
drawing does not run on every tps

### DISCUSSION THREAD

**Discussion by jdemaria22** [hajimehoshi/ebiten#2541] [Q&A]
Hi guys, i have a question about the Draw() function in ebitten.

sometimes the function Update() executing two or three times before de Draw() function.

For example, this is a sequence of my program with 165 TPS . 

Update()
Update()
Update()
Draw()
Update()
Draw()
......

Someone can help me? 

Sorry for my english

**Comment by hajimehoshi** [comment]
This is usual. For one frame (a vsync), multiple Updates can be called, or sometimes Update might not be called. Ebitengine tries to call Update the specified TPS times per second on average.

**Reply by jdemaria22** [reply]
So, i need to call Update and Draw every tick, 
First Update and after Draw.
What is the config for do that?

**Reply by hajimehoshi** [reply]
What about `ebiten.SetTPS(ebiten.SyncWithFPS)`?

In the first place, I'm not sure why you want to call Update exactly once every frame.

**Reply by jdemaria22** [reply]
Update, find the information about the position of circles and images for draw, in 165 tps, the problem is that the elements that I draw flicker. And the TPS and FPS show me 165.

**Reply by hajimehoshi** [reply]
I don't quite understand what is going on. Could you show me your code?

**Reply by jdemaria22** [reply]
yes, of course


func main() {
	NAME = strconv.Itoa(rand.Intn(RANDOM))
	ebiten.SetWindowSize(screenWidth, screenHeight)
	ebiten.SetWindowTitle(NAME)
	ebiten.SetWindowResizingMode(ebiten.WindowResizingModeDisabled)
	ebiten.SetTPS(ebiten.SyncWithFPS)
	ebiten.SetScreenClearedEveryFrame(true)
	ebiten.SetWindowDecorated(false)
	ebiten.SetWindowFloating(true)
	ebiten.SetVsyncEnabled(true)
	game := &Game{}
	option := ebiten.RunGameOptions{}
	option.GraphicsLibrary = ebiten.GraphicsLibraryAuto
	option.InitUnfocused = false
	option.ScreenTransparent = true
	option.SkipTaskbar = true
	err := ebiten.RunGameWithOptions(game, &option)
	if err != nil {
		log.Fatal(err)
	}
}

type Game struct {
}

func (g *Game) Update() error {
	module.Update() //this update de information (values to draw)
	return nil
}

func (g *Game) Draw(screen *ebiten.Image) {
	scripts.DrawChamps(screen) // method to draw
	ebitenutil.DebugPrint(screen, fmt.Sprintf("TPS: %0.2f\nFPS: %0.2f", ebiten.ActualTPS(), ebiten.ActualFPS()))
}

func (g *Game) Layout(outsideWidth, outsideHeight int) (int, int) {
	return 1920, 1080
}

**Reply by hajimehoshi** [reply]
I can't tell why the 'flickering' happens from this code. I think there is an issue in the module or the script part.
